### PR TITLE
[Docs] Separate UWP build instructions

### PR DESCRIPTION
**Pull request by matthuisman**
## Description

newer versions of the SDK in VS2017 no longer have the Desktop C++ and UWP separate entries.
Also, i tried 15063 & 16299 and both give errors.
When i changed to 17764 - everything was fine

## Motivation and context

spent a few hours hitting syntax error generated from the SDK (Windows Kits\10\Include)
with the current docs, users may try use the older versions that have the desktop and uwp entries separate (15063 & 16299)

## How has this been tested?

n/a

## What is the effect on users?

less trouble building on windows

## Screenshots (if appropriate):
image

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [X] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by thexai** [comment]
All SDK on this picture are obsolete now and NOT valid to build UWP.

image

As states install notes for Visual Studio 2019 is required **10.0.18362.0**:

> **Under SDKs, libraries, and frameworks section select**
> 
> - Windows 10 SDK (10.0.18362.0) (if compiling for UWP or UWP-ARM)

Only is need this SDK number if compiling compiling for UWP or UWP-ARM (for Windows desktop can be used any posterior SDK number  i.e.  10.0.19041)  

What happens is that SDK (10.0.18362.0) is not included with Visual Studio 2017 installer and so "select the latest version" is not correct. And 10.0.x.0 is also not correct as **only** 10.0.18362.0 can be used.

Install notes for VS 2019 are correct.
Install notes for VS 2017 are incorrect and should be indicated same as 2019:

"Windows 10 SDK (10.0.18362.0) (if compiling for UWP or UWP-ARM)"
https://developer.microsoft.com/en-US/windows/downloads/sdk-archive/

I no longer have VS 2017 installed and I cannot verify anything but Jenkins still uses VS 2017 and this SDK had to be installed as a separate download and it is currently working.

File: docs/README.Windows.md [hunk 1]
```diff
@@ -98,8 +98,7 @@ Start the VS2017 installer and click `Individual components`.
 * Under **Development activities** select
   * `Visual Studio C++ core features`
 * Under **SDKs, libraries, and frameworks** select
-  * `Windows 10 SDK (10.0.x.0) for Desktop C++ [x86 and x64]`
-  * `Windows 10 SDK (10.0.x.0) for UWP: C++`
+  * `Windows 10 SDK (10.0.x.0)` (select the latest version)
```
Comment (thexai) [inline docs/README.Windows.md:101]:
```suggestion
  * `Windows 10 SDK (10.0.18362.0) (if compiling for UWP or UWP-ARM)
```

Comment (matthuisman) [reply docs/README.Windows.md:101]: that tells me if im compiling for x64 - i dont need any sdk...

when I was trying to compile, I was just doing x64 (not uwp)
 15063 & 16299  didnt work but 17764 did

So maybe we need both?
*  `Windows 10 SDK (10.0.x.0)` (if compiling for x86 and x64)
* `Windows 10 SDK (10.0.18362.0) (if compiling for UWP or UWP-ARM)

**Comment by matthuisman** [comment]
what do you think about separating UWP instructions?

Something like this: https://github.com/matthuisman/xbmc/blob/windows_doc/docs/README.md
(maybe someone could make a UWP logo same as the IOS / macos as UWP doesn't really have a recognised logo)

for UWP, just have instructions for VS 2019 as 2017 doesnt include the required SDK.
Makes it all a bit simpler.

Separating them out like this makes them much easier to follow IMO

If UWP ever gets dropped, then just remove the files for it.

File: docs/README.Windows.md (hunk 1 above)
Comment (thexai) [reply docs/README.Windows.md]: > that tells me if im compiling for x64 - i dont need any sdk?

No, this tells us that the _default SDK_ is fine. Installing VS always installs a default SDK unless a custom install is used.

The default SDK in VS 2017 is 10.0.17763.0
The default SDK in VS 2019 is 10.0.19041.0

Both are fine to compile x64 desktop.

The instructions only say that additional components need to be installed only in the case of UWP.

Additional components are:

> MSVC v142 - VS 2019 C++ ARM build tools (Latest) (if compiling for UWP-ARM)

> Windows 10 SDK (10.0.18362.0) (if compiling for UWP or UWP-ARM)


Then when compiling UWP and additional components are installed you ends up with two SDK:

**Visual Studio 2017**
SDK 10.0.17763.0 (*) +  SDK 10.0.18362.0

or

**Visual Studio 2019**
SDK  10.0.19041.0 (*)  + SDK  10.0.18362.0

(*) = default

And when compiling x64 more recent SDK installed are auto picked: 
VS 2017 --> 10.0.18362.0  
VS 2019 --> 10.0.19041.0

On the contrary, when compiling UWP | UWP-32 | UWP-ARM the target SDK is always used:
VS 2017 --> 10.0.18362.0 
VS 2019 --> 10.0.18362.0

**Comment by thexai** [comment]
I don't think there is a need to separate anything since many requirements are common (CMake, JRE). Separating it makes it more difficult to maintain because it will almost always be necessary to update two documents instead of one.

**Comment by thexai** [comment]
Maybe @DaveTBlake's point of view is important here as I think he is still using VS2017 and I was able to identify if there are things that are wrong in the documentation.

File: docs/README.Windows.md [hunk 2]
```diff
@@ -77,40 +77,25 @@ Default options are fine.
...
 * Under **Desktop & Mobile** section select
   * `Desktop development with C++`
-  * `Universal Windows Platform development` (if compiling for UWP or UWP-ARM)
 
 Click in **Individual components** select
-* Under **Compilers, build tools and runtimes** section select
-  * `MSVC v142 - VS 2019 C++ ARM build tools (Latest)` (if compiling for UWP-ARM)
 * Under **SDKs, libraries, and frameworks** section select
-  * `Windows 10 SDK (10.0.18362.0)` (if compiling for UWP or UWP-ARM)
+  * `Windows 10 SDK (10.0.18362.0)`
```
Comment (thexai) [inline docs/README.Windows.md:83]: Read my comments is not need select any additional individual component for compiling x64 desktop

Windows 10 SDK (10.0.19041.0) is fine and is always installed by default.

Comment (matthuisman) [reply docs/README.Windows.md:83]: Thing is. This is not when using default settings. This is specifically selecting only what is needed to build. To reduce size. If sdk isn't mentioned then I'd not select that and it won't work.

**Comment by matthuisman** [comment]
I'm just going to close this and leave it upto the team. Just remember these docs should be accessible and make sense for a simple user to follow so they can get involved. A dev would be in a position to figure out stuff when it goes wrong.

Also VS is huge. I don't even use its ide. I just want to install the bare minimum (not defaults) to get a build built to test changes.

**Comment by thexai** [comment]
> I just want to install the bare minimum (not defaults) to get a build built to test changes.

You can install SDK 10.0.18362.0 from external download and "untick" defaults SDKs provided by VS 2017 installer or VS 2019 installer.  10.0.18362.0 is valid for compile both UWP and desktop and valid for use with VS 2017 / VS 2019.

Instructions are easy to follow: some components are checked by default. Instructions only says additional component's that need "tick" for UWP (assuming you not changed the defaults).

Of course, advanced users can optimize the installation space and use other options. But these users don't need to follow the manual anyway.

**Comment by thexai** [comment]
>  i tried 15063 & 16299 and both give errors.

Where is it written that these SDKs must/may be used?

**Comment by matthuisman** [comment]
Lets back it up a bit and focus on 2017 for starters.

1) No where in current docs does it say to keep the defaults and install these additional
    It reads to me as "just install these components".
2) With 2017, none of the defaults are actually needed. Only what is listed in the docs.
   Which is great - no bloat just to build.
3) Current docs say for 2017
`Windows 10 SDK (10.0.x.0) for Desktop C++ [x86 and x64]`
 `Windows 10 SDK (10.0.x.0) for UWP: C++`
 (The "for UWP" isn't a Kodi note - its actually what is meant to be listed next to the SDK)
 
 Now look at the below image of what SDKS are available in 2017
 https://user-images.githubusercontent.com/6225961/146109714-70265828-d6b7-4186-bdbc-f3d59b2ee6cb.png
 
The user would select 16299 as that's the latest which has "for Desktop..." in it's name.
 
 Obviously with the newer SDK, the have changed the name because they have merged in the UWP stuff.
 
 So my first PR attempt was to change this so at least the name in the docs matches the name of the SDK.
 
 But then you said18362 is minimum needed for UWP builds.
 So that changes the game again.
 Now do we tell them not to install SDK via VS Installer and manually instally it.
 Or tell them to installl 7763 if not doing UWP, but if UWP then download manually.
 The instructions start getting quite messy then, so I thought separating them would be cleaner.

**Comment by thexai** [comment]
> and focus on 2017 for starters.

For starting users, there is no reason to prefer VS 2017 over VS 2019. Honestly everything is more complicated in VS 2017 and lacking in support / updates.

So it seems like a waste of time to have to change the entire documentation format expressly to support VS 2017.

What I think is more convenient is to remove VS 2017 from the instructions (maybe leave a note in the manual commands saying that it can be used).

VS 2019 includes all the SDKs that are needed in the installer itself:

Captura de pantalla 2021-12-21 103859

The instructions are simpler: for desktop you only have to mark an option in "Workloads". You do not need to change anything in "individual components":

Captura de pantalla 2021-12-21 103728

It is not good that Kodi recommends installing old software lacking updates. It should be understood that VS 2019 is the recommended version (at a minimum) and that VS 2022 can also be used.

**Comment by matthuisman** [comment]
I'm all for removing 2017 if it's just going to make things more confusing or waste time. I feel the guide should be aimed at first time users doing their first install. So if it's easier to say go with 2019, then great. It's just I had 2017 installed when I noticed the sdk differences and ended up spendinf more time trying different sdks than just switching to 2019. I think it's better to have a single choice that works than having multiple choices where one doesn't or wastes time

**Pull request closed without merging**